product_template_multi_link/models/product_template_link.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-Today GRAP (http://www.grap.coop).
+# @author Sylvain LE GAL <https://twitter.com/legalsylvain>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProductTemplateLink(models.Model):
+    _name = 'product.template.link'
+    _order = 'product_template_id, linked_product_template_id'
+
+    _TYPE_SELECTION = [
+        ('cross_sell', 'Cross-Sell'),
+        ('up_sell', 'Up-Sell'),
+    ]
+
+    product_template_id = fields.Many2one(
+        string='Source Product', comodel_name='product.template',
+        required=True, ondelete='cascade')
+
+    product_template_image_small = fields.Binary(
+        related='product_template_id.image_small')
+
+    linked_product_template_id = fields.Many2one(
+        string='Linked Product', comodel_name='product.template',
+        required=True, ondelete='cascade')
+
+    linked_product_template_image_small = fields.Binary(
+        related='linked_product_template_id.image_small')
+
+    type = fields.Selection(
```

Comment:
maybe better to use link_type as type already mean something in python

Reply:
Thanks for your review. type renamed in link_type.

regard.